### PR TITLE
Add definition to customize disabled quick starts

### PR DESCRIPTION
**Pull request by sahil143**
fixes: https://issues.redhat.com/browse/ODC-5566

Extends the Console CRD to add `quickStarts.disabled` option to allow cluster admins to disable quick starts.

Based on enhancement proposal: https://github.com/openshift/enhancements/pull/715

### CODE REVIEW

File: operator/v1/types_console.go [hunk 1]
```diff
@@ -171,6 +175,14 @@ type DeveloperConsoleCatalogCategory struct {
 	Subcategories []DeveloperConsoleCatalogCategoryMeta `json:"subcategories,omitempty"`
 }
 
+// QuickStarts allow cluster admins to customize available quick starts
+type QuickStarts struct {
+	// disabled is a list of names of quick starts that are not shown to the users
```
Comment (spadgett) [inline operator/v1/types_console.go:180]: We should make it clear that these are the names ConsoleQuickStart resources. I would probably use the full PascalCase name here and in the comment above to make it clear it's referencing a specific CRD.

Eventually, we'll need to update this to handle the name annotation used for translations, but I would not mention that yet until we've implemented translations for quick starts.

Comment (sahil143) [reply operator/v1/types_console.go]: @spadgett Updated as per the suggestion.

File: operator/v1/types_console.go [hunk 2]
```diff
@@ -171,6 +175,14 @@ type DeveloperConsoleCatalogCategory struct {
 	Subcategories []DeveloperConsoleCatalogCategoryMeta `json:"subcategories,omitempty"`
 }
 
+// QuickStarts allow cluster admins to customize available ConsoleQuickStart resources
+type QuickStarts struct {
+	// disabled is a list of names of ConsoleQuickStart resources that are not shown to the users
```
Comment (spadgett) [inline operator/v1/types_console.go:180]:
```suggestion
	// disabled is a list of ConsoleQuickStart resource names that are not shown to users.
```

File: operator/v1/types_console.go [hunk 3]
```diff
@@ -119,6 +119,10 @@ type ConsoleCustomization struct {
 	// +kubebuilder:validation:Optional
 	// +optional
 	ProjectAccess ProjectAccess `json:"projectAccess,omitempty"`
+	// quickStarts allows customization of available ConsoleQuickStart resources in console
```
Comment (spadgett) [inline operator/v1/types_console.go:122]:
```suggestion
	// quickStarts allows customization of available ConsoleQuickStart resources in console.
```

**Review by spadgett — COMMENTED** [review]
Sorry, I'm being a little picky about the API doc since these messages are visible to the user in console and through `oc explain`.

**Comment by spadgett** [comment]
Looks like you have an error in the verify job.

**Review by spadgett — APPROVED** [review]
/lgtm

**Pull request merged**